app/questionnaire/value_source_resolver.py

```diff
@@ -100,12 +100,35 @@ def _resolve_list_item_id_for_value_source(
             else None
         )
 
+    def _resolve_dynamic_answers(
+        self,
+        answer_id: str,
+    ) -> Generator[AnswerValueTypes, None, None]:
+        # Type ignore: the answer block will exist at this stage
+        question = self.schema.get_block_for_answer_id(answer_id).get("question", {})  # type: ignore
+
+        if dynamic_answers := question.get("dynamic_answers"):
+            values = dynamic_answers["values"]
+            if values["source"] == "list":
+                for list_item_id in self.list_store[values["identifier"]]:
+                    if answer_value := self._get_answer_value(
+                        answer_id=answer_id, list_item_id=list_item_id
+                    ):
+                        yield answer_value
+
     def _resolve_answer_value_source(
         self, value_source: Mapping
     ) -> ValueSourceEscapedTypes | ValueSourceTypes:
+        """resolves answer value by first checking if the answer is dynamic whilst not in a repeating section,
+        which indicates that it is a dynamic answer resolving to a list. Otherwise, retrieve answer value as normal.
+        """
         list_item_id = self._resolve_list_item_id_for_value_source(value_source)
         answer_id = value_source["identifier"]
 
+        # if not in a repeating section and the id is for a list of dynamic answers, then return the list of values
+        if not list_item_id and self.schema.is_answer_dynamic(answer_id):
+            return list(self._resolve_dynamic_answers(answer_id))
```

Comment:
Correct me if I am wrong here, but if the only usage of `_resolve_dynamic_answers()` is here and we want a list then make that method return a list.

Reply:
Fair point - updated this, thanks 👍

Reply:
I mean, generator is more universal but given that we only use it once. It's not some blocker change.

Reply:
Yeah I think you're right, yielding looked quite clean but theres no point in the generator if its its just going to be turned straight into a list